forwards_paper.tex

```diff
@@ -658,43 +624,51 @@ \subsection*{Simulation benchmarks}
 
 
 %%%%%%%%%%%%%%%%%%%%%%
-\section*{Conclusion}
-
-
-This works great and very fast.
-
-Having results in a tree sequence is really good downstream too:
-The tree sequence encoding is compact but it is also efficient to process.
-Many algorithms to compute statistics of interest for population genetics
-are naturally expressed in terms of tree topologies. For example, the
-pairwise nucleotide diversity $\pi$, is defined as the average number of
-differences between sequences in the sample. Computing $\pi$ directly
-from observed sequence data requires $O(n^2 m)$ time, since we must
-compare all pairs of samples at every site. However, given the topology of
-the trees at every site and the locations of mutations much better algorithms
-are possible. Using the fact that we can count the number of samples below
-a given node efficiently using previously described tree sequence
-algorithms~\citep{kelleher2016efficient}, the time required to compute $\pi$
-becomes roughly $O(n \log n + m)$.
+\section*{Discussion}
+
+In this paper, we have shown that storing pedigrees
+and associated recombination events
+in a forwards-time simulation
+not only results in having available a great deal more information about the simulation,
+but also can speed up the simulation by orders of magnitude.
+To make this feasible, 
+we have described how to efficiently store this information in tables,
+and have described a fundamental algorithm for simplificication of tree sequences.
+Conceptually, recording of genealogical and recombination events
+can happen independently of the details of simulation;
+for this reason, we provide a well-defined and well-tested API in C and in python
+for use in other code bases.
+
+The tree sequences produced by default by this method
+are very compact, storing genotype \emph{and} genealogical information
+in a small fraction of the space taken by a compressed VCF file.
+The format is also highly efficient to process,
+leading to advantages for downstream analysis as well.
+This is because many algorithms to compute statistics of interest for population genetics
+are naturally expressed in terms of tree topologies,
+and so can be quickly computed from the trees underlying the tree sequence format. 
+For example, pairwise nucleotide diversity $\pi$, is the average density of
+differences between sequences in the sample. 
+To compute this directly from sequence data at $m$ sites in $n$ samples
+requires computing allele frequencies, taking $O(nm)$ operations.
```

Comment:
previously this said $n^2 m$, which would be to compute the matrix of differences.